nautobot/docs/installation/region-and-site-data-migration-guide.md

```diff
@@ -0,0 +1,408 @@
+        # Point the location field to the corresponding "Site" LocationType Location
+        # with the same name.
+        example_model.location = example_model.site.migrated_location
+    ExampleModel.objects.bulk_update(example_models, ["location"], 1000)
+
+class Migration(migrations.Migration):
+    dependencies = [
+        # The dcim migration creates the Site Type and Region Type Locations that
+        # your data models are migrating to. It has to be run **before** this migration.
+        ("dcim", "0030_migrate_region_and_site_data_to_locations"),
+        ("example_app", "0007_add_location_field_to_example_model"),
+    ]
+    operations = [
+        migrations.RunPython(
+            # Execute the function
+            code=migrate_example_model_data_to_locations,
+            reverse_code=migrations.operations.special.RunPython.noop,
+        )
+    ]
+```
+
+### Remove Site/Region Related Fields from Migrated Data Models
+
+After the data migration is successful, we need to remove the `site`/`region` fields from your data model so that Nautobot will be able to remove the `Site` and `Region` models. Note that we need to remove those attributes in a separate migration file from the previous one, as it's never a good practice to combine data migrations and schema migrations in the same file. You can do this by simply removing the `site`/`region` attributes from your model class:
+
+```python
+# models.py
+class ExampleModel(OrganizationalModel):
+    # note that site field is gone
+    location = models.ForeignKey(to="dcim.Location", on_delete=models.PROTECT, blank=True, null=True)
+    name = models.CharField(max_length=20, help_text="The name of this Example.")
+...
+```
+
+After removing the `site` attribute, make the migration file by running `nautobot-server makemigrations [app_name] -n [migration_name]`, for example:
+
+```shell
+nautobot-server makemigrations example_app -n remove_site_field_from_example_model
+```
+
+The migration file might look like this:
+
+```python
+# Generated by Django 3.2.17 on 2023-02-22 17:09
+# 0009_remove_site_field_from_example_model.py
+
+from django.db import migrations
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('example_app', '0008_migrate_example_model_data_from_site_to_location'),
+    ]
+
+    operations = [
+        migrations.RemoveField(
+            model_name='examplemodel',
+            name='site',
+        ),
+    ]
+```
+
+!!! important
+    Before you apply this migration, you have to add a `run_before` attribute in this migration file to make sure that you remove `site`/`region` fields before `Site` and `Region` models themselves are removed. **Without it, your migration files might be out of order and your app will not start**.
+
+```python
+    # Ensure this migration is run before the migration that removes Region and Site Models
+    run_before = [
+        # TODO we need to change the name after we reorder migration files
```

Comment:
When will this be done?

Reply:
After we merge #3313 I think, added that to the TODO comment.